mm2src/coins/lp_coins.rs

```diff
@@ -3595,29 +3615,142 @@ impl MmCoinStruct {
+                        burn_destination: DexFeeBurnDestination::BurnAccount,
+                    };
+                }
+            }
+        }
+        DexFee::Standard(dex_fee)
+    }
+
+    /// Returns dex fee discount if KMD is traded
+    pub fn dex_fee_rate(base: &str, rel: &str) -> MmNumber {
+        let fee_discount_tickers: &[&str] = match var("MYCOIN_FEE_DISCOUNT") {
+            Ok(_) => &["KMD", "MYCOIN"],
+            Err(_) => &["KMD"],
+        };
+        if fee_discount_tickers.contains(&base) || fee_discount_tickers.contains(&rel) {
+            // 1/777 - 10%
+            BigRational::new(9.into(), 7770.into()).into()
+        } else {
+            BigRational::new(1.into(), 777.into()).into()
         }
     }
 
+    /// Drops the dex fee in KMD by 25%. This cut will be burned during the taker fee payment.
+    ///
+    /// Also the cut can be decreased if the new dex fee amount is less than the minimum transaction amount.
+    fn calc_burn_amount_for_op_return(dex_fee: &MmNumber, min_tx_amount: &MmNumber) -> (MmNumber, MmNumber) {
+        // Dex fee with 25% cut
+        let new_fee = dex_fee * &MmNumber::from(Self::BURN_CUT);
+        if &new_fee >= min_tx_amount {
+            // Use the max burn value, which is 25%.
+            let burn_amount = dex_fee - &new_fee;
+            // we don't care if burn_amount < dust as any amount can be sent to op_return
+            (new_fee, burn_amount)
+        } else if dex_fee >= min_tx_amount {
+            // Burn only the exceeding amount because fee after 25% cut is less than `min_tx_amount`.
+            let burn_amount = dex_fee - min_tx_amount;
+            (min_tx_amount.clone(), burn_amount)
+        } else {
+            (dex_fee.clone(), MmNumber::from(0))
+        }
+    }
+
+    /// Drops the dex fee in non-KMD by 25%. This cut will be sent to an output designated as 'burn account' during the taker fee payment
+    /// (so it cannot be dust).
+    ///
+    /// The cut can be set to zero if any of resulting amounts is less than the minimum transaction amount.
+    fn calc_burn_amount_for_burn_account(dex_fee: &MmNumber, min_tx_amount: &MmNumber) -> (MmNumber, MmNumber) {
+        // Dex fee with 25% cut
+        let new_fee = dex_fee * &MmNumber::from(Self::BURN_CUT);
+        let burn_amount = dex_fee - &new_fee;
+        if &new_fee >= min_tx_amount && &burn_amount >= min_tx_amount {
+            // Use the max burn value, which is 25%. Ensure burn_amount is not dust
+            return (new_fee, burn_amount);
+        }
+        // If the new dex fee is dust set it to min_tx_amount and check the updated burn_amount is not dust.
+        let burn_amount = dex_fee - min_tx_amount;
+        if &new_fee < min_tx_amount && &burn_amount >= min_tx_amount {
+            // actually currently burn_amount (25%) < new_fee (75%) so this never happens. Added for a case if 25/75 will ever change
+            return (min_tx_amount.clone(), burn_amount);
```

Comment:
Why? Users will pay the same original dex fee amount. We just change its proportion so that the dex fee share is min-amount (non-dust) and ensure the burn output is not dust too.

Reply:
Yes, you are right, I missed this part
```
let burn_amount = dex_fee - min_tx_amount;
```